src/components/CookieModal.tsx

```diff
@@ -0,0 +1,51 @@
+import { useCallback, useEffect, useState } from "react";
+
+const CookieModal = () => {
+    const [cookieValue, setCookieValue] = useState<string | null>(null);
+    const [loading, setLoading] = useState<boolean>(true);
+
+    const handleButtonClick = useCallback((choice: string) => {
+        localStorage.setItem('userCookie', choice);
+        setCookieValue(choice);
+    }, []); 
+    
+    useEffect(() => {
+        const userCookie = localStorage.getItem('userCookie');
+        setCookieValue(userCookie);
+        setLoading(false);
+    }, []);
+
+    if (loading) return null;
```

Comment:
We can combine condition here for loading and coockieValue no?

Reply:
Yeah we can. Updated that,